Archive/Imo/Imo1982Q1.lean

```diff
@@ -0,0 +1,125 @@
+/-
+Copyright (c) 2024 Alex Brodbelt. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex Brodbelt, Violeta Hernández
+-/
+
+import Mathlib.Tactic.NormNum
+import Mathlib.Data.PNat.Basic
+
+/-!
+# IMO 1982 Q1
+
+The function $f(n)$ is defined for all postive integers $n$ and takes on
+non-negative integer values. Also, for all $m$, $n$
+
+$f (m + n) - f(m) - f(n) = 0 \text{ or } 1$
+$f(2) = 0, f(3) > 0, and f(9999) = 3333.$
+
+Determine $f(1982)$.
+
+The solution is based on the one at the
+[Art of Problem Solving](https://artofproblemsolving.com/wiki/index.php/1982_IMO_Problems/Problem_1)
+website.
+
+We prove the helper lemmas:
+1. $nf(m) \leq f(mn)$.
+2. $f(m) + f(n) \leq f(m + n)$.
+3. $a f(a) + c f(d) \leq f(a \cdot b + c \cdot d)$ (derived from 1. and 2.).
```

Comment:
```suggestion
1. $nf(m) \leq f(mn)$ (`superhomogeneous`)
2. $f(m) + f(n) \leq f(m + n)$ (`superadditive`)
3. $a f(a) + c f(d) \leq f(a \cdot b + c \cdot d)$ (`superlinear`, derived from 1. and 2.)
```
Note that you'll have to reorder 1 and 2 if you employ my suggestion of proving `superhomogeneous` using `superadditive`.

Reply:
I think I will keep it as it is.

Reply:
It doesn't have to be exactly how I wrote it, but I think it is useful linking the relevant proofs alongside their mathematical statements.

Reply:
Wait, are you referring to the docstring, or the reordering of lemmas? If you mean the former that's fine, all I really wanted was for the Lean lemmas to be mentioned alongside their mathematical statements. If you mean the latter then I really don't see your objection, especially when the new proof becomes so much simpler.

Reply:
implemented the suggestions (: